hypervisor/include/dm/vpci.h

```diff
@@ -184,4 +184,28 @@ int32_t vpci_assign_pcidev(struct acrn_vm *tgt_vm, struct acrn_assign_pcidev *pc
 int32_t vpci_deassign_pcidev(struct acrn_vm *tgt_vm, struct acrn_assign_pcidev *pcidev);
 struct pci_vdev *vpci_init_vdev(struct acrn_vpci *vpci, struct acrn_vm_pci_dev_config *dev_config, struct pci_vdev *parent_pf_vdev);
 
+static inline bool is_pci_io_bar(struct pci_vbar *vbar)
+{
+        return ((!vbar->is_mem64hi) && (vbar->bar_type.io_space.indicator == 1U));
+}
+
+static inline bool is_pci_mem_bar(struct pci_vbar *vbar)
+{
+        return ((!vbar->is_mem64hi) && ((vbar->bar_type.mem_space.indicator == 0U)));
+}
+
+static inline bool is_pci_reserved_bar(struct pci_vbar *vbar)
+{
+        return (is_pci_mem_bar(vbar) && ((vbar->bar_type.mem_space.mem_type & 0x1U) == 0x1U));
```

Comment:
We need to add check of !is_mem64hi here.

vbar->bar_type.mem_space.indicator == 0U && (vbar->bar_type.mem_space.mem_type & 0x1U) == 0x1U  && !is_mem64hi

Reply:
Yes, that is caused by issues of is_pci_mem_bar(), I thought mem64hi bar is not memory bar, that is wrong. will fix it